src/components/UI/NavBar/NavBar.jsx

```diff
@@ -2,39 +2,66 @@ import styles from "./NavBar.module.css";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { faPhotoFilm } from "@fortawesome/free-solid-svg-icons";
 import { useEffect, useState } from "react";
+import { Link, NavLink } from "react-router-dom";
+import { Button } from "../Button/Button";
+import { useAuth } from "../../../context/auth-context";
+import { useUserData } from "../../../context/userdata-context";
 export const NavBar = () => {
+  const { authState, authStateDispatch } = useAuth();
+  const { userDataDispatch } = useUserData();
   const [isScrolled, setIsScrolled] = useState(false);
-  useEffect(()=>{
-    const handleScroll = () =>{
-      if(window.scrollY > 0){
+  useEffect(() => {
+    const handleScroll = () => {
+      if (window.scrollY > 0) {
         setIsScrolled(true);
       } else {
         setIsScrolled(false);
       }
-    }
+    };
 
-    window.addEventListener("scroll", handleScroll)
-    return () =>{
+    window.addEventListener("scroll", handleScroll);
+    return () => {
       window.removeEventListener("scroll", handleScroll);
-    }
-
-  },[]);
+    };
+  }, []);
   return (
     <>
-      <nav className={`${styles["nav"]} ${!isScrolled && styles["transparent"]}`}>
-        <div className={styles["nav-brand"]}>
+      <nav
+        className={`${styles["nav"]} ${!isScrolled && styles["transparent"]}`}
+      >
+        <Link to="/" className={styles["nav-brand"]}>
           <FontAwesomeIcon icon={faPhotoFilm} size="2x" />
           MeTube
-        </div>
+        </Link>
         <menu className={styles["nav-links"]}>
-          <li>Home</li>
-          <li>My list</li>
-          <li>Contact Us</li>
+          <li>
+            <NavLink to="/">Home</NavLink>
```

Comment:
hey bro i just came to know that we have to use NAVLINK if we want apply style to active link if not we can use link
even i did this mistake so if you didn't require style so use LINK